src/vidpf.rs

```diff
@@ -0,0 +1,735 @@
+// Copyright (c) 2024 Cloudflare, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//! Implementation of a Verifiable Incremental Distributed Point Function (VIDPF).
+//!
+//! The VIDPF construction is specified in [[draft-mouris-cfrg-mastic]] and builds
+//! on techniques from [[MST23]] and [[CP22]] to lift an IDPF to a VIDPF.
+//!
+//! [CP22]: https://doi.org/10.1007/978-3-031-06944-4_6
+//! [MST23]: https://eprint.iacr.org/2023/080
+//! [draft-mouris-cfrg-mastic]: https://datatracker.ietf.org/doc/draft-mouris-cfrg-mastic/01/
+
+use std::{
+    borrow::Borrow,
+    error::Error,
+    iter::zip,
+    marker::PhantomData,
+    ops::{Add, BitAnd, BitXor, ControlFlow, Sub},
+};
+
+use rand_core::RngCore;
+use subtle::{Choice, ConditionallyNegatable, ConditionallySelectable};
+
+use crate::{
+    codec::Decode,
+    field::{FieldElement, FieldElementExt},
+    vdaf::xof::{Seed as XofSeed, Xof},
+};
+
+/// Creates a new instance of a VIDPF.
+pub fn new_vidpf<P, R>(sec_param: usize, prng_ctor: P, bind_info: Vec<u8>) -> VidpfInstance<P, R>
+where
+    P: GetPRNG,
```

Comment:
fyi: Rust convention is to not capitalize acronyms (unlike Go)
```suggestion
    P: GetPrng,
```